Controllers/AbstractControllers/AbstractCatalogController.php

```diff
@@ -0,0 +1,176 @@
+{
+    protected AbstractCatalogModel $catalogModel;
+    protected AbstractCatalogView $catalogView;
+    protected ?AbstractBaseMsgModel $msgModel = null;
+    protected ?IDataManagement $serverInfo = null;
+
+    public function __construct(AbstractCatalogModel $catalogModel, AbstractCatalogView $catalogView)
+    {
+        $this->catalogModel = $catalogModel;
+        $this->catalogView = $catalogView;
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function showAction(): void
+    {
+        try {
+            if ($this->validateRequest()) {
+                $this->prepareRedirect();
+            } else {
+                $this->catalogModel->setMsgModel($this->getMsgModel('request'));
+                $data = $this->catalogModel->catalog();
+                $options = $this->catalogView->getOptions('Каталог', 'catalog.phtml', $data);
+                $this->catalogView->render($options);
+            }
+        } catch (\Exception $exception) {
+            $this->catchException($exception);
+        }
+    }
+
+    /**
+     * @param string $uriType
+     * @return AbstractBaseMsgModel
+     * @throws \Exception
+     */
+    protected function getMsgModel(string $uriType = 'default'): AbstractBaseMsgModel
+    {
+        switch ($uriType) {
+            case 'referer' :
+                return $this->getMsgModelByReferer();
+            case 'request' :
+                return $this->getMsgModelByRequest();
+            case 'default' :
+                $this->msgModel = MsgModelsFactory::getMsgModel('default');
+                return $this->msgModel;
+            default :
+                throw new \Exception('Unknown MsgModel type in AbstractUserController :' . " '$uriType'");
+        }
+    }
+
+    /**
+     * @param string $option
+     * @return string
+     * @throws \Exception
+     */
+    protected function getRefererOption(string $option): string
+    {
+        return $this->getServerInfo()->getRefererOption($option);
+    }
+
+    /**
+     * @param string $option
+     * @return string
+     * @throws \Exception
+     */
+    protected function getRequestOption(string $option): string
+    {
+        return $this->getServerInfo()->getRequestOption($option);
+    }
+
+    /**
+     * @return string
+     * @throws \Exception
+     */
+    protected function getRequestUserType(): string
+    {
+        return $this->getRequestOption('user_type');
+    }
+
+    /**
+     * @return string
+     * @throws \Exception
+     */
+    protected function getRequestController(): string
+    {
+        return $this->getRequestOption('controller');
+    }
+
+    /**
+     * @return string
+     * @throws \Exception
+     */
+    protected function getRequestAction(): string
+    {
+        return $this->getRequestOption('action');
+    }
+
+    /**
+     * @return string
+     * @throws \Exception
+     */
+    protected function getRefererAction(): string
```

Comment:
is there a way to avoid lot's of getRequestController getRequestAdminType getRequestAction